### PR TITLE
[Tabs] Sling Resource Merger properties break item add button

### LINKED ISSUE #618: [Tabs] Sling Resource Merger properties breaks item add button
## Bug Report

**Current Behavior**
I have extended the core Tabs component by overlaying it in my app with sling:resourceSuperType="core/wcm/components/tabs/v1/tabs". I have modified the tabs.html and written my own clientlib to make it function like an accordion, as such I do not want authors to be able to set the activeItem as I want all accordions to be collapsed by default. Adding sling:hideChildren="properties" on the tab/items node in the dialog successfully achieves the hiding of the 'properties' tab however the 'Add' button on the Items tab is no longer functional. Clicking Add button no longer opens the Add New Components popover. To further prove this is a bug I also tried added sling:hideResource="{Boolean}true" on the properties node itself and experience the same behaviour. Likewise adding sling:hideResource on the activeItems node also breaks the Add component functionality.

eg:
`<tabs jcr:primaryType="nt:unstructured"
                sling:resourceType="granite/ui/components/coral/foundation/tabs"
                maximized="{Boolean}true">
                <items jcr:primaryType="nt:unstructured" sling:hideChildren="properties">`

**Expected behavior/code**
I would not expect hiding either the 'properties' tab nor any of its child properties to have any affect on the functionality of the Add component button. The Add component button on the Items tab should still function as normal.

**Environment**
- AEM 6.4 SP3
- Core Components version  2.4.0
- JRE version (e.g. `Java(TM) SE Runtime Environment build 1.8.0_211-b12

**Possible Solution**

**Additional context / Screenshots**
Add any other context about the problem here. If applicable, add screenshots to help explain.
Screen Shot 2019-06-07 at 1 51 20 pm

### PR DESCRIPTION
**Pull request by richardhand**
- ensure the children editor works if the properties tab and its elements aren't available

| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | `Fixes #618` 
| Patch: Bug Fix?          | 👍
| Minor: New Feature?      |
| Major: Breaking Change?  |
| Tests Added + Pass?      |
| Documentation Provided   | (code comments and or markdown)
| Any Dependency Changes?  |
| License                  | Apache License, Version 2.0

### CODE REVIEW

File: content/src/content/jcr_root/apps/core/wcm/components/tabs/v1/tabs/clientlibs/editor/js/tabs.js [hunk 1]
```diff
@@ -32,17 +32,19 @@
             var activeSelect = tabsEditor.querySelector(activeSelectSelector);
             var activeItem = tabsEditor.querySelector(activeItemSelector);
 
-            Coral.commons.ready(childrenEditor, function() {
-                updateActiveSelect(childrenEditor, activeSelect, activeItem);
-            });
+            if (childrenEditor && activeSelect && activeItem) {
+                Coral.commons.ready(childrenEditor, function() {
+                    updateActiveSelect(childrenEditor, activeSelect, activeItem);
```
Comment (jckautzmann) [inline content/src/content/jcr_root/apps/core/wcm/components/tabs/v1/tabs/clientlibs/editor/js/tabs.js]: I would prefer to encapsulate the checks within the `updateActiveSelect` function.

Comment (richardhand) [reply content/src/content/jcr_root/apps/core/wcm/components/tabs/v1/tabs/clientlibs/editor/js/tabs.js]: We should still check for each element here, as we bind events to them:
```
activeSelect.on("change", function() {
                    activeItem.value = activeSelect.value;
});
```

Comment (jckautzmann) [reply content/src/content/jcr_root/apps/core/wcm/components/tabs/v1/tabs/clientlibs/editor/js/tabs.js]: to have a more robust function I would still add the checks in the function.

Comment (richardhand) [reply content/src/content/jcr_root/apps/core/wcm/components/tabs/v1/tabs/clientlibs/editor/js/tabs.js]: Updated in 4838abb6ddcd2345bcc307daebfca42d9b9fafb4

**Pull request merged**